lib/fog/orchestration/openstack/requests/create_stack.rb

```diff
@@ -28,6 +31,19 @@ def create_stack(arg1, arg2 = nil)
             }.merge(arg2.nil? ? {} : arg2)
           end
 
+          # Templates should always:
+          #  - be strings
+          #  - contain URI references instead of relative paths.
+          # Passing :template_url may not work well with `get_files` and remote `type`:
+          #  the python client implementation in shade retrieves from :template_uri
+          #  and replaces it with :template.
+          #  see https://github.com/openstack-infra/shade/blob/master/shade/openstackcloud.py#L1201
+          #  see https://developer.openstack.org/api-ref/orchestration/v1/index.html#create-stack
+          file_resolver = Fog::Orchestration::Util::RecursiveHotFileLoader.new(options[:template] || options[:template_url], options[:files])
```

Comment:
better this way
```
if options[:files].nil?
  file_resolver = Fog::Orchestration::Util::RecursiveFileLoader.new(options[:template] || options[:template_url])
  files = file_resolver.get_files
  options[:files] = files unless files.empty?
  options[:template] = file_resolver.template
end

Reply:
:files reintroduced after gitter discussion